lib/base-compiler.ts

```diff
@@ -523,7 +523,7 @@ export class BaseCompiler {
     // Returns a list of additional options that may be required by some backend options.
     // Meant to be overloaded by compiler classes.
     // Default handles the GCC compiler with some debug dump enabled.
-    optionsForBackend(backendOptions, outputFilename) {
+    optionsForBackend(backendOptions: Record<string, any>, outputFilename: string): string[] {
```

Comment:
We should probably get a list of all of the options at some point

Reply:
You are right. Will try on Monday

Reply:
There are lots of possible options, so lets move it to an eventual different PR